presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -529,4 +530,11 @@ else if (!previousPairsWereEqual) {
         // the input is sorted, but the algorithm has still failed
         throw new IllegalArgumentException("failed to find a group ending");
     }
+
+    @Override
+    public void close()
+            throws Exception
+    {
+        driverWindowInfo = Optional.of(windowInfo.build());
```

Comment:
I see your point, it COULD be useful to see that a WF was going smoothly in the beginning of the execution but went south due to skew at some point, although currently all our use cases (analysis of customers performance problems) require info gathered in `explain analyze` in the end of the query. It's not worth to pollute network and use CPU to compute intermediate results we don't want, IMO.
If we ever end up with a desire to read intermediate stats for WFs, we can enable them then.

Also, AD the first statement, I'm not sure what value without additional instrumentation it gives us to know that WF was executing fine for 3 minutes and started having troubles for the remaining e.g. 7 minutes of its 10 minute-time execution.